### PR TITLE
Chat toggles for ghost.

### PR DESCRIPTION
**Pull request by eswordthecat**

### CODE REVIEW

**Comment by eswordthecat** [comment]
Can be pulled if success.

File: code/modules/mob/dead/observer/say.dm [hunk 1]
```diff
@@ -27,10 +27,22 @@
 	return "[pick("whines", "cries", "spooks", "complains", "drones", "mutters")], \"[text]\"";
 
 /mob/dead/observer/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq)
-	if(radio_freq)
-		speaker = speaker.GetSource()
+	if (isnull(client))
+		return
+
+	if (isnull(radio_freq))
+		if (!(client.prefs.toggles & CHAT_GHOSTEARS))
```
Comment (ComicIronic) [inline code/modules/mob/dead/observer/say.dm]: Does this make it impossible to hear what's going on on-screen when you have ghostears disabled?

Comment (eswordthecat) [inline code/modules/mob/dead/observer/say.dm]: If you're ghost, yes.

Comment (eswordthecat) [inline code/modules/mob/dead/observer/say.dm]: Same with ghost radio.

File: code/modules/mob/dead/observer/say.dm [hunk 2]
```diff
@@ -27,10 +27,22 @@
...
-	if(radio_freq)
-		speaker = speaker.GetSource()
+	if (isnull(client))
+		return
+
+	if (isnull(radio_freq))
+		if (!(client.prefs.toggles & CHAT_GHOSTEARS))
+			return
+	else
+		if (!(client.prefs.toggles & CHAT_GHOSTRADIO))
```
Comment (ComicIronic) [inline code/modules/mob/dead/observer/say.dm]: Doesn't this make it so that any radio that transmits on-screen is silent to a ghost?

Comment (eswordthecat) [inline code/modules/mob/dead/observer/say.dm]: If you toggle it off at preferences.

File: code/modules/mob/dead/observer/say.dm (hunk 1 above)
Comment (ComicIronic) [inline code/modules/mob/dead/observer/say.dm]: But that's not what Ghostears does, I'm pretty sure.
It's a toggle between global and local chat, not chat and nochat.

Comment (eswordthecat) [inline code/modules/mob/dead/observer/say.dm]: setup.dm:720

**Comment by eswordthecat** [comment]
Okay, done boss. Can we remove the tag nowwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww?

**Pull request merged**